src/main/java/com/hierynomus/smbj/session/Session.java

```diff
@@ -93,6 +103,9 @@ private Share connectTree(final String shareName) {
         SmbPath smbPath = new SmbPath(remoteHostname, shareName);
         logger.info("Connection to {} on session {}", smbPath, sessionId);
         try {
+            if (connection.getConfig().isDFSEnabled()) {
```

Comment:
There should be a better way than referring to these static methods.

Either the session should be wrapped, or get a `DFSResolver`, or the `SmbPath` should be extended. Don't have a nice solution yet, but this is not the right one I think.